docker-compose.yml

```diff
@@ -1,33 +1,55 @@
-version: '2'
+# This file is used to bring up a single node
+# development and test environment with Bigchain,
+# MongoDB and Tendermint
+#
+# 'bdb-base' service is only used to build the 'bigchaindb'
+# base image which is then used by 'bdb' service to build a
+# development image. Having a base image with all dependencies
+# already installed helps reduce build time of development image
+# significantly
+
+version: '3'
 
 services:
   mdb:
     image: mongo:3.4.3
     ports:
       - "27017"
     command: mongod
-
+  bdb-base:
+    image: bigchaindb
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        backend: localmongodb
+    volumes:
+      - ./bigchaindb:/usr/src/app/bigchaindb
+      - ./tests:/usr/src/app/tests
   bdb:
+    depends_on:
+      - mdb
+      - tendermint
     build:
       context: .
       dockerfile: Dockerfile-dev
-      args:
-        backend: mongodb
     volumes:
       - ./bigchaindb:/usr/src/app/bigchaindb
       - ./tests:/usr/src/app/tests
-      - ./docs:/usr/src/app/docs
-      - ./k8s:/usr/src/app/k8s
-      - ./setup.py:/usr/src/app/setup.py
-      - ./setup.cfg:/usr/src/app/setup.cfg
-      - ./pytest.ini:/usr/src/app/pytest.ini
-      - ./tox.ini:/usr/src/app/tox.ini
```

Comment:
Why are these removed?

For instance, `pytest.ini` and `setup.cfg` control `pytest` and are needed when developing and testing.

`docs` are an essential part of development and also need to be mounted.

`setup.py` may be needed to install a new package, and thus it can be useful to mount it.

Reply:
@shahbazn  can you comment on this?

Reply:
was just using `docker-compose.tendmint.yml` as the source of truth and replacing the docker-compose.yml with `docker-compose.tendermint.yml` and thats what is reflected here. Now that `docker-compose.tendmint.yml` have been updated will update this